inspect.go

```diff
@@ -0,0 +1,166 @@
+package enmime
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"mime"
+	"net/textproto"
+	"strings"
+
+	"github.com/jhillyerd/enmime/internal/coding"
+	"github.com/pkg/errors"
+)
+
+func HumanHeadersOnly(b []byte) (map[string]string, error) {
+	b = clean(b)
+	tr := textproto.NewReader(bufio.NewReader(bytes.NewReader(b)))
+	headers, err := tr.ReadMIMEHeader()
+	switch errors.Cause(err) {
+	case nil, io.EOF:
+	// carry on, io.EOF is expected
+	default:
+		return nil, err
+	}
+	bs := bufio.NewScanner(bufio.NewReader(bytes.NewReader(b)))
+	res := map[string]string{}
+	for bs.Scan() {
+		line := bs.Text()
+		if strings.HasPrefix(line, " ") || strings.HasPrefix(line, "\t") {
+			continue
+		}
+		i := strings.Index(line, ":")
+		if i == -1 {
+			continue
+		}
+		header := textproto.CanonicalMIMEHeaderKey(line[:i])
+		if len(headers[header]) == 0 {
+			// somethings up, we should have already processed all of these, so why are we trying to fetch from an empty list, did we miscount?
```

Comment:
Comment is a little confusing, maybe mention it's populated by ReadMIMEHeader?

Reply:
this was removed